BLSQ/iaso
# Ia 1520 mod instance details

The code of instance details and its related speed dial was messy due to details being a class component (thus unable to use hooks)

## Self proof reading checklist



## Changes

- Refactor Instance `details` as TS function component (courtesy of @olethanh )
- Refactor `SpeedDialInstances`:
    - use a hook to generate each action in order to isolate the different logics
    - separate the networking code
    - reorganize the component fro readablity
    - reduce the payload passed to org units API


## How to test

- enable the `SHOW_LINK_INSTANCE_REFERENCE` feature flag on the account you're using for testing
- go to submissions and try linking/unlinking instances
- It should be compared to staging to spot any regression

## Print screen / video


https://user-images.githubusercontent.com/38907762/190139270-91e9fde2-aae8-4944-99ba-54d3dde1801c.mov




## Notes

@hakifran  Can you please test the linking feature? 

There's more refactoring that can be done, notably:
- Remove redux from `SpeedDialInstance`
- Refactor `ActionTableColumnComponent` in a similar fashion to `SpeedDialInstance` (This one will really help when we refactor the API as well)

But I'm running out of available time for this refactor, so I suggest we open separate tickets for this.

@olethanh I didn't refactor `SpeedDialAction` as it would have entailed some refactoring in org units as well, and I left the additional url params for the linking feature as removing them was breaking
